packages/react-composites/src/composites/CallComposite/components/SidePane/SidePaneProvider.tsx

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React, { createContext, useCallback, useContext, useEffect, useMemo } from 'react';
+
+/** @private */
+export type InjectedSidePaneProps =
+  | undefined
+  | {
+      headerRenderer?: () => JSX.Element;
+      contentRenderer?: () => JSX.Element;
+      sidePaneId: string;
+      // Useful to ensure the side pane renders the content of the override even if the side pane is closed.
+      // This avoids remounting the content when the side pane is opened again.
+      hidden?: boolean;
+    };
+
+interface SidePaneProps {
+  /** Side pane header */
+  setHeaderRenderer: React.Dispatch<React.SetStateAction<(() => JSX.Element) | undefined>>;
+  headerRenderer?: () => JSX.Element;
+  /** Side pane content */
+  setContentRenderer: React.Dispatch<React.SetStateAction<(() => JSX.Element) | undefined>>;
+  contentRenderer?: () => JSX.Element;
+  /** tracking open state of the side pane */
+  setActiveSidePaneId: React.Dispatch<React.SetStateAction<string | undefined>>;
+  activeSidePaneId: string | undefined;
+  /** Support injecting content into the side pane from extrnal sources (e.g. CallWithChatComposite's Chat) */
+  setOverrideSidePane: React.Dispatch<React.SetStateAction<InjectedSidePaneProps>>;
+  overrideSidePane: InjectedSidePaneProps;
+}
+
+const defaultSidePaneProps: SidePaneProps = {
+  setHeaderRenderer: () => () => <></>,
+  setContentRenderer: () => () => <></>,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  setActiveSidePaneId: () => {},
+  activeSidePaneId: undefined,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  setOverrideSidePane: () => {},
+  overrideSidePane: undefined
+};
+
+/**
+ * Context for side pane interaction
+ *
+ * @private
+ */
+export const SidePaneContext = createContext<SidePaneProps>(defaultSidePaneProps);
+
+/**
+ * Props to LocalizationProvider
+ *
+ * @private
+ */
+export type SidePaneProviderProps = {
+  children: React.ReactNode;
+};
+
+/**
+ * Provider to provide localized strings for this library's composites.
+ *
+ * @private
+ */
+export const SidePaneProvider = (props: SidePaneProviderProps): JSX.Element => {
+  const [headerRenderer, setHeaderRenderer] = React.useState<(() => JSX.Element) | undefined>();
```

Comment:
Is it possible to avoid adding global changing state into provider, it's a huge review push back to do so long time ago - it's another format of global state - our centralized state not single source of truth of global anymore. The conclusion of the discussion was we could use Provider for global static, but we shouldn't put any dynamic changing state there for either debuggability or code structure. It feels this is opening a new pattern that brings us back to that design

if we feel that is a need to add these states into adapter, I will 100% vote for that